libs/components/src/next/form/BaseFormControl.ts

```diff
@@ -0,0 +1,135 @@
+import { PropertyDeclarations } from 'lit';
+import {
+    FormControlMixin,
+    maxLengthValidator,
+    minLengthValidator,
+    patternValidator,
+    requiredValidator,
+} from '@open-wc/form-control';
+import { LitElement } from 'lit';
+import { submit } from '@open-wc/form-helpers';
+import { maxValueValidator, minValueValidator } from './validators';
+import { ERROR_MESSAGE_CUSTOM_TAG } from './error-message/vl-error-message.component';
+import { BaseLitElement } from '@domg-wc/common-utilities';
+
+export abstract class BaseFormControl extends FormControlMixin(BaseLitElement) {
+    id = '';
+    protected name = '';
+    protected label = '';
+    protected block = false;
+    protected required = false;
+    protected isInvalid = false;
+
+    static formControlValidators = [
+        requiredValidator,
+        minLengthValidator,
+        maxLengthValidator,
+        minValueValidator,
+        maxValueValidator,
+        patternValidator,
+    ];
+
+    static shadowRootOptions = { ...LitElement.shadowRootOptions, delegatesFocus: true };
+
+    static get properties(): PropertyDeclarations {
+        return {
+            id: { type: String, reflect: true },
+            name: { type: String, reflect: true },
+            label: { type: String, reflect: true },
+            block: { type: Boolean, reflect: true },
+            required: { type: Boolean, reflect: true },
+            isInvalid: { type: Boolean, state: true },
+        };
+    }
+
+    connectedCallback(): void {
+        super.connectedCallback();
+
+        this.addEventListener('keydown', this.onKeydown);
+        this.addEventListener('invalid', this.onInvalid);
+
+        if (!this.label) {
+            this.label = this.internals.labels[0]?.innerText || '';
+        }
+    }
+
+    disconnectedCallback(): void {
+        super.disconnectedCallback();
+
+        this.removeEventListener('keydown', this.onKeydown);
+        this.removeEventListener('invalid', this.onInvalid);
+    }
+
+    updated(changedProperties: Map<string, unknown>): void {
+        super.updated(changedProperties);
+
+        if (!changedProperties.has('isInvalid')) {
+            this.isInvalid = false;
+            this.hideErrorMessages();
+        }
+    }
+
+    abstract get validationTarget(): HTMLElement | undefined | null;
+
+    resetFormControl(): void {
+        this.isInvalid = false;
+        this.hideErrorMessages();
+        this.dispatchEvent(new Event('reset'));
+    }
+
+    private onKeydown = (event: KeyboardEvent): void => {
```

Comment:
`onKeyDown` lijkt met al specifieke implementatie voor een form-control waarin getypt wordt wat niet per sé het geval zal zijn voor bv. `vl-date-picker` of `vl-upload`, maar wel voor een groot deel van de form-control's dus beetje dubbel

Ook voor en `vl-textarea` zou ik mss overwegen dit dan parametriseerbaar te maken (of het op `Enter` gaat submitten)

zou dit dan mss definiëren ter hoogte van de specifieke afnemende klasse/component (of toch nog een input tussenklasse maken)

is een voorstel; kunnen dit bespreken